src/context.ts

```diff
@@ -59,30 +56,26 @@ function createContext<
     }
     return useStore(
       store,
-      selector as StateSelector<TState, StateSlice>,
+      selector as StateSelector<ExtractState<S>, StateSlice>,
       equalityFn
     )
   }
 
-  const useStoreApi = (): {
-    getState: CustomStoreApi['getState']
-    setState: CustomStoreApi['setState']
-    subscribe: CustomStoreApi['subscribe']
-    destroy: CustomStoreApi['destroy']
-  } => {
+  const useStoreApi = (): S => {
     const store = useContext(ZustandContext)
     if (!store) {
       throw new Error(
         'Seems like you have not used zustand provider as an ancestor.'
       )
     }
     return useMemo(
-      () => ({
-        getState: store.getState,
-        setState: store.setState,
-        subscribe: store.subscribe,
-        destroy: store.destroy,
-      }),
+      () =>
+        ({
+          getState: store.getState,
+          setState: store.setState,
+          subscribe: store.subscribe,
+          destroy: store.destroy,
+        } as S),
```

Comment:
`as S` seems wrong here. We don't even need any type assertion?

Reply:
Hmm if I remember correctly this was so that the inferred type shows `S` and not `{ getState: S["getState"], ... }` etc. But then we have already annotated the return type with `:S` so this probably is not required, will recheck.

Reply:
Also look like I've accidently deleted `return { Provider, useStore: useBoundStore, useStoreApi }`? Because it in the [base branch](https://github.com/pmndrs/zustand/blob/v4/src/context.ts#L90-L94), but then why the diff doesn't show I deleted it? I'm so confused rn, probably I should sleep xD

Reply:
This should be `{ getState: S["getState"], ... }` because we want to avoid the function call usage.

Reply:
> Yes, because the goal of this is to avoid misusage of hooks. S contains a hook, and we want to not expose it.

Gotcha, fixed it although I added a type `WithoutCallSignaure<S>` so that the tooltip is more readable, otherwise it'll show `{ getState: S["getState"], ... }`, I'll revert it you don't like it.